src/hb-ot-shape-complex-use.cc

```diff
@@ -451,19 +452,37 @@ reorder_syllable (hb_buffer_t *buffer, unsigned int start, unsigned int end)
 
   hb_glyph_info_t *info = buffer->info;
 
-#define BASE_FLAGS (FLAG (USE_B) | FLAG (USE_GB))
+#define POST_BASE_FLAGS (FLAG (USE_FM) | \
+			 FLAG (USE_FAbv) | \
+			 FLAG (USE_FBlw) | \
+			 FLAG (USE_FPst) | \
+			 FLAG (USE_MAbv) | \
+			 FLAG (USE_MBlw) | \
+			 FLAG (USE_MPst) | \
+			 FLAG (USE_MPre))
 
   /* Move things forward. */
   if (info[start].use_category() == USE_R && end - start > 1)
   {
-    /* Got a repha.  Reorder it to after first base, before first halant. */
+    /* Got a repha.  Reorder it towards the end, but before the first post-base
+     * glyph. */
     for (unsigned int i = start + 1; i < end; i++)
-      if ((FLAG_UNSAFE (info[i].use_category()) & (BASE_FLAGS)) || is_halant (info[i]))
+    {
+      bool is_post_base_glyph = (FLAG_UNSAFE (info[i].use_category()) & POST_BASE_FLAGS) ||
+				info[i].use_category() == USE_VBlw ||
```

Comment:
Not using flags for these because they are over 32?  Maybe we can add 64bit flags?

Reply:
Yes, that’s why. A wider `FLAG_UNSAFE` would be convenient. Do you want me to add one for this PR?

Reply:
Yes please.  Call it FLAG64 or whatever needs to be.